packages/contracts/contracts/B.Protocol/BAMM.sol

```diff
@@ -163,7 +166,25 @@ contract BAMM is CropJoinAdapter, PriceFormula, Ownable {
         return n.mul(uint(10000 + bps)) / 10000;
     }
 
-    function getSwapEthAmount(uint lusdQty) public view returns(uint ethAmount, uint feeEthAmount) {
+    function compensateForLusdDeviation(uint ethAmount) public view returns(uint newEthAmount) {
+        uint chainlinkDecimals;
+        uint chainlinkLatestAnswer;
+
+        // get current decimal precision:
+        chainlinkDecimals = lusd2UsdPriceAggregator.decimals();
+
+        // Secondly, try to get latest price data:
+        (,int256 answer,,,) = lusd2UsdPriceAggregator.latestRoundData();
+        chainlinkLatestAnswer = uint(answer);
+
+        // adjust only if 1 LUSD > 1 USDC. If LUSD < USD, then we give a discount, and rebalance will happen anw
```

Comment:
Comment mentions both USDC and USD, assume only one is correct.

Reply:
yeah, it suppose to be USD (not USD*C*). Thanks.